projects/developer_experience.md

```diff
@@ -0,0 +1,104 @@
+
+Convenience as a SIG was initially proposed by [Ted
+Young](https://github.com/tedsuo) back in 2021. It was postponed to prioritize
+completion of signal specs and stabilization. A document was written at this
+time describing the intent to do user research and a brainstorming session,
+https://docs.google.com/document/d/1bjiw5L4E8FztRtfsy4Nu9A6qSP-BJlK6BBzl6Rea-Ds/edit#heading=h.b7qckyjgu28i.
+This brainstorm resulted in a list of some existing convenience functions
+languages have and proposed ones, like a `start` which takes the tracer from the
+active span in the current context, and proposals for defining annotations for
+easier instrumentation.
+
+### Current challenges
+
+We know users can find the API and SDK to be complex and daunting, the challenge
+is to collect actionable feedback on this issues. That information is spread
+across various languages' issues, blog posts and other online commentary.
+Performing an end user survey will help collect some of this knowledge into a
+central place that is actionable.
+
+When developer experience issues are encountered and addressed they are
+currently done in the various language implementations individually and this can
+lead to drift between different implementations and with the spec. An example
+would be how to handle marking a code block as `untraced` which has
+implementations in at least Java, Ruby and Javascript, each done differently,
+named differently and with no specification. This means users working across
+languages experience different results for similar features and other language
+SIG's aren't necessarily even aware of the pain point that their users may
+actually be experiencing.
+
+### Goals, objectives, and requirements
+
+With the 3 main signals going stable the time is right to circle back and
+address issues users may have faced over the years they've now been using
+OpenTelemetry.
+
+This project aims to identify developer experience issues with using
+OpenTelemetry, to select 3 that are concerns which can be addressed through
+updates to the specification and resolve them.
+
+## Deliverables
+
+The first deliverable will be the collection of experience in dealing with
+developer experience issues by each existing language SIG. This means not only
+additions to the API/SDK or libraries developed to enhance the experience for
+users but any that may be planned or are being thought about because of a
+frequent request from their users. A report of the findings will be shared with
+the whole community.
+
+Utilizing the knowledge gained from the first deliverable on potential areas of
+improvement the next deliverable will be an end user survey. Assuming the agreement and
+participation of the End User SIG this will be done in cooperation with them. A
+write-up of the results will follow, along with the group's decision on the top 3
+issues for the group to work on.
+
+Prototypes of resolutions will be developed and, depending on the scope of the
```

Comment:
This initiative should be larger than just the API shape of the SDK. Developer experience is not just about writing code that consumes the SDK. Its about the development lifecycle - discovery/learning, tooling, debugging/diagnostics, testing, CI/CD & app deployment. 

Examples:
- How do I enable the built in telemetry to each platform & if there isn't built-in telemetry getting it built
- How do I discover the names/sources for what needs to be enabled to get telemetry and understand the scope of how much data will be generated by the source
- How do I view my telemetry at design-time
  - How do I have different sampling for the inner loop so I get all the telemetry, but have sensible defaults for production deployment
- How can I verify if the right telemetry is being emitted as part of unit tests
- Can I use telemetry as a way of monitoring system behavior as part of integration tests?
- How does OPs know what telemetry is available for the app, know what needs to be configured to enable it and what the dials are to control the volume/granularity of telemetry produced.

Reply:
This is an extremely broad list, many of which would be extremely difficult to provide guidance for on a cross-project basis. I'm not saying they're bad ideas, but to take one -- for example, 'verifying if the right telemetry is being emitted as part of unit tests' -- would involve quite a bit of design and research on testing libraries, scaffolding, and patterns across many languages in order to discover the cross-product of necessary hooks that could be recommended for inclusion at a specification level.

Reply:
I'm happy with the current proposed scope for this SIG, and it can grow in the future depending on how the needs evolve. I'm marking this as resolved, but please open an issue if you think this needs to be discussed further.